test/integration/viewing_history_test.rb

```diff
@@ -0,0 +1,78 @@
+require "test_helper"
+require "capybara/rails"
+
+class ViewingHistoryIntegrationTest < ActionDispatch::IntegrationTest
```

Comment:
As per our convo, I feel like this could probably be a good fit for a Cucumber test, in the same way the Notes feature is tested. But I'm open to other opinions!